src/material/menu/menu.spec.ts

```diff
@@ -1147,6 +1147,7 @@ describe('MatMenu', () => {
     fixture.detectChanges();
 
     fixture.componentInstance.trigger.menuOpened.subscribe(() => {
+      flush();
```

Comment:
@crisbeto I'm confused about this test. I understand that for the `openMenu` method, there is no requirement on when the first element of the list is focused. It has to be focused in some time after calling openMenu.

For the `menuOpened` event, is there a requirement that the first element must be focused by the time the `menuOpened` event fires?

To avoid a race condition with the `_focusFirstItem` method, this test adds a flush. In application code, the developer would have to wait for `onStable` or do a `setTimeout`. There is still a way for developers to move focus inside the `open` event callback, but not at the time the `open` event is fired.

Does this PR meet the requirements for the behavior of `menuOpened`?

Reply:
This is checking that we don't override the consumer's focus, if they moved it into the menu themselves before we tried to focus the first item. That's why I was proposing the `if (!menu.contains(document.activeElement)` check.